src/components/Ratings/Ratings.tsx

```diff
@@ -0,0 +1,268 @@
+      onSelect(idx);
+    }
+  };
+
+  setRef = ref => {
+    this.iconRefs.push(ref);
+  };
+
+  _renderContent = () => <StarIcon />;
+
+  _renderInputOptions = () => {
+    const { inputOptions, value } = this.props;
+    const { currentHovered, currentFocus } = this.state;
+    const currentValue = currentFocus ? currentFocus : currentHovered;
+
+    return (
+      <div className={styles.labelList}>
+        {currentValue ? (
+          <span
+            data-hook={RATINGS_DATA_HOOKS.InputOption}
+            className={styles.inputOption}
+          >
+            {inputOptions[currentValue - 1]}
+          </span>
+        ) : (
+          <span
+            data-hook={RATINGS_DATA_HOOKS.InputOptionCurrent}
+            className={classNames(
+              styles.inputOption,
+              styles.inputOptionCurrent,
+            )}
+          >
+            {inputOptions[value - 1]}
+          </span>
+        )}
+      </div>
+    );
+  };
+
+  _renderRatingInfo = () => {
+    const { ratingDisplay, countDisplay } = this.props;
+    const info = [];
+
+    if (ratingDisplay) {
+      info.push(ratingDisplay);
+    }
+
+    if (countDisplay) {
+      info.push(countDisplay);
+    }
+
+    return (
+      <div
+        data-hook={RATINGS_DATA_HOOKS.RatingInfo}
+        className={styles.ratingInfo}
+      >
+        {info.join(' | ')}
+      </div>
+    );
+  };
+
+  render() {
+    const {
+      value,
+      onSelect,
+      disabled,
+      error,
+      size,
+      mode,
+      inputOptions,
+      layout,
+      ratingDisplay,
+      countDisplay,
+      ...rest
+    } = this.props;
+    const content = this._renderContent();
+    const ratingList = Array.from(new Array(5));
+    const ratingListLength = ratingList.length;
+    const showInputOptions = inputOptions.length && mode === Mode.Input;
+    const showRatingInfo =
+      (ratingDisplay || countDisplay) && mode === Mode.Display;
+
+    return (
+      <div
+        {...styles('root', { disabled, error, size, mode, layout }, rest)}
+        {...this.getDataAttributes()}
+        onKeyDown={this.handleKeyDown}
+        onBlur={this.handleUnfocus}
+        data-hook={this.props['data-hook']}
+      >
+        <div className={styles.iconList}>
+          {ratingList.map((_el, idx: number) => {
+            const humanOrder = ratingListLength - idx;
+            const checked = humanOrder <= value;
+            const ariaLabel = inputOptions.length
+              ? inputOptions[4 - idx]
+              : (5 - idx).toString();
+
+            return (
+              <span
+                data-hook={RATINGS_DATA_HOOKS.IconWrapper}
+                key={idx}
+                {...styles(styles.icon, { checked })}
+                onMouseEnter={() => this.handleHoverIcon(humanOrder)}
+                onMouseLeave={this.handleUnhoverIcon}
+                ref={this.setRef}
+                tabIndex={-1}
+              >
+                <CoreRadio
```

Comment:
I think our main problem lies here:
When we focus on the component the `span` above ☝️ is taking the focus, and if we `TAB` again, the CoreRadio component catches the focus.
We need to pass a `tabIndex={-1}` to the CoreRadio, which I'm not sure it currently supports.
Which means we will need to create a simple PR for `wix-ui-core` in order to add the `tabIndex` property.